### PR TITLE
 Add block level drop-down menu

### PR DESCRIPTION
**Pull request by kaviththiranga**

### CODE REVIEW

File: composer/packages/diagram/src/views/components/block-dropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    public componentWillReceiveProps(nextProps: BlockDropdownProps) {
+        this.setState({
+            isHovered: nextProps.active
+        });
+    }
+
+    public render() {
+        const { triggerPosition } = this.props;
+        const { isHovered } = this.state;
+        return <Fragment>
```
Comment (pahans) [inline composer/packages/diagram/src/views/components/block-dropdown.tsx]: seems `Fragment` unnecessary here.

File: composer/packages/diagram/src/views/components/block.tsx [hunk 2]
```diff
@@ -2,16 +2,46 @@
...
+        const hoverRectBBox = {
+            height: h, width : w + leftMargin, x: x - leftMargin,  y
+        };
         statements.push(DiagramUtils.getComponents(model.statements));
         return (
-            <g className="panel">
+            <g
+                className="panel block"
+            >
                 {statements}
```
Comment (pahans) [inline composer/packages/diagram/src/views/components/block.tsx]: `statements` is a blank array?

File: composer/packages/diagram/src/views/components/block-dropdown.tsx (hunk 1 above)
Comment (kaviththiranga) [reply composer/packages/diagram/src/views/components/block-dropdown.tsx]: will fix

File: composer/packages/diagram/src/views/components/block.tsx (hunk 2 above)
Comment (kaviththiranga) [reply composer/packages/diagram/src/views/components/block.tsx]: nope, see statements.push(DiagramUtils.getComponents(model.statements));

File: composer/packages/diagram/src/views/components/block-dropdown.tsx (hunk 1 above)
Comment (kaviththiranga) [reply composer/packages/diagram/src/views/components/block-dropdown.tsx]: @pahans fixed

**Pull request merged**